clang/lib/StaticAnalyzer/Checkers/MallocChecker.cpp

```diff
@@ -1877,16 +1923,43 @@ static bool didPreviousFreeFail(ProgramStateRef State,
   return false;
 }
 
+static void printOwnershipTakesList(raw_ostream &os, CheckerContext &C,
+                                    const Expr *E) {
+  if (const CallExpr *CE = dyn_cast<CallExpr>(E)) {
```

Comment:
When I advocated for early-returns, I wanted something like this:
```c++
const auto *CE = dyn_cast<CallExpr>(E);
if (!CE)
  return;
const FunctionDecl *FD = CE->getDirectCallee();
if (!FD)
  return;
  ...
```
On the same token, the `FD->hasAttrs()` appears to be not strictly necessary as in that case the loop simply wouldn't take any iterations at all, right?